pkgs/swift2objc/lib/src/transformer/transformers/transform_struct.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../../ast/_core/interfaces/declaration.dart';
+import '../../ast/_core/shared/parameter.dart';
+import '../../ast/declarations/built_in/built_in_declaration.dart';
+import '../../ast/declarations/compounds/class_declaration.dart';
+import '../../ast/declarations/compounds/members/initializer.dart';
+import '../../ast/declarations/compounds/members/property_declaration.dart';
+import '../../ast/declarations/compounds/struct_declaration.dart';
+import '../../parser/_core/utils.dart';
+import '../_core/unique_namer.dart';
+import '../transform.dart';
+import 'transform_method.dart';
+import 'transform_property.dart';
+
+ClassDeclaration transformStruct(
```

Comment:
Again this looks pretty much identical to `transformClass`. Try to have `transformStruct` and `transformClass` wrap a shared `transformCompound` function. See how much code duplication you can eliminate.

Reply:
Same as parsing, refactored the whole two functions.